packages/react-core/src/hooks/useIsSignedIn.ts

```diff
@@ -0,0 +1,45 @@
+import { Hub } from 'aws-amplify/utils';
+import { useEffect, useState } from 'react';
+import { HubCallback } from '@aws-amplify/core';
+import { getCurrentUser } from 'aws-amplify/auth';
+
+/**
+ * Returns `false`:
+ * - if user has no credentials cached in browser storage
+ * - if user signs out of account
+ *
+ * Returns `true`:
+ * - if user has cached credentials
+ * - if user signs in to account
+ */
+export default function useIsSignedIn(): boolean {
+  const [isSigned, setIsSigned] = useState<boolean>(false);
+
+  useEffect(() => {
+    getCurrentUser()
+      .then(() => {
+        setIsSigned(true);
+      })
+      .catch(() => {
+        setIsSigned(false);
+      });
+  }, []);
```

Comment:
It can since there is a case in which the end user is already signed in but getCurrentUser is loading or down. 

Since it is an async function, perhaps it would be better to keep useIsSignedIn as undefined rather than assuming the user is not signed in.